### DISCUSSION TITLE
REGION is enormous ! Triggers everything!

### DISCUSSION THREAD

**Discussion by Minglarn** [blakeblackshear/frigate#3029] [Ask A Question]
Hi..
Trying to understand why regions are so enormous.. 
When a car passes by the REGION is triggering everything on my parking lot. Is there a settings for this or what could be the cause?
Even people walking triggers the car closest to the wall on my parking. ... 

Se below.. 
![region_enormus](https://user-images.githubusercontent.com/58695742/160644509-d211ec29-a678-4a17-92b1-d52aa5fed289.gif)

**Comment by NickM-27** [comment]
What is the detect resolution of your camera? I believe regions should be a minimum of 320x320 which is the resolution the models use for object detection.

**Reply by Minglarn** [reply]
It's 640x480 ...

**Reply by NickM-27** [reply]
Okay, that explains why the regions are relatively large since the camera image is smaller.

**Reply by Minglarn** [reply]
That means that low-stream is useless for my type detection? 
Well then, seems that I've to go back to high-stream again... pity. This increases the CPU usage by 15-20% ..

**Reply by NickM-27** [reply]
Well, not necessarily. Instead of using the high res stream, you could simply change the `width` and `height` under `detect` (maybe like double it) which will have ffmpeg just resize the image of the low res stream. Will use more CPU but likely much less than moving to the high res stream.

**Reply by Minglarn** [reply]
Didn't make any difference... Same giant REGION... Guess i've to go back to hi-stream instead... 
Doubled the size of lo-res stream, but the quality not acceptable.

**Reply by NickM-27** [reply]
Mind sharing your config before and after? It definitely should be making a difference.

**Reply by Minglarn** [reply]
Hum... That's odd... Wen't back to hi-res stream, and the regions are as big when using the lo-res stream... 
Will try to record it and show it...

**Reply by NickM-27** [reply]
Just watched on my camera and could see the region grow / shrink so it must be taking the region and resizing it to 320x320, which makes a lot more sense and I was being silly before. The overall resolution does still affect it though as I am pretty sure the region will have a minimum size of 320x320 (or whatever the model size is if a different one is used)

edit: my assumption was correct: https://github.com/blakeblackshear/frigate/blob/08f573aaa5ec0b9bf3d4b40388942254448b902a/frigate/video.py#L533

**Comment by Minglarn** [comment]
The resolution is now at 2560x1920... 
https://user-images.githubusercontent.com/58695742/160652273-f22b142d-12d2-4e3b-bd80-25a332ce1174.mp4

**Reply by NickM-27** [reply]
Okay yeah so I was partially correct, the regions are smaller than before with the higher res. The reason the region gets bigger once the person gets over by the car is because the car is detect and then a larger region is used to keep track of where the car is.

**Reply by NickM-27** [reply]
With that cleared up, what exactly is the issue you are trying to solve? Are there events occurring that you want to ignore or what is happening that needs to be avoided?

**Reply by Minglarn** [reply]
I was trying to switch to the lo-stream on all my cams, but this one is reacting to all moving objects even when zones/mask are defined..
But as you mentioned, the region is as smallest 320x320 and thats to big for the 640x480 stream. 
When set to hi-res stream the regions are acceptable at daytime, but I do get a lot of false detections night time as flares from passing cars reflect on my cars. 

The cam is to far away from the street to be useful at lo-res with Frigate.

**Reply by NickM-27** [reply]
How are your zones defined? I'm not sure why you'd have detections since the vehicles should remain in the zones even with the boxes changing positioning some.

**Reply by Minglarn** [reply]
The zones are define as below:
![image](https://user-images.githubusercontent.com/58695742/160673576-79d908d4-b291-432f-aef6-94ba9b83c7ac.png)

**Comment by blakeblackshear** [comment]
There is a bunch of confusion here about regions, etc. The size of the region is based on the size of the motion area as well as any surrounding objects. It tries to group up nearby areas of interest and adjust the region to reduce the total number of times that object detection needs to run on a single frame. The region is **not** restricted to 320x320, but it won't allow a region to be smaller than 320x320 as it would have to scale up the image and that is just a waste. If the area of interest is **larger** than 320x320, frigate will resize it down to 320x320 to run object detection. 

Changing the resolution doesn't really have any impact on the region size other than limiting the minimum size.

I can assure you that reducing the region size is not your goal here. You have cars parked in your driveway, and one thing or another is going to cause them to be detected. It could be a passing cloud or a car driving by. With stationary object detection, frigate should already know about these cars and they should already have bounding boxes. You **want** frigate to constantly show bounding boxes for these cars while they are in the driveway as this helps prevent frigate from accidentally getting confused and assuming that a car driving by entered your driveway. 

After reading through this I don't see where you have described an actual problem. In the video you posted originally, did you get a notification when you weren't expecting one? Did you get an event saved that you weren't expecting? All I see is a video of frigate working exactly as designed.

**Reply by NickM-27** [reply]
Thanks, good to understand how that part is working behind the scenes as I definitely was off. I think I eventually got most of the way there, but makes sense resolution changing isn't a good approach.
